File: spark-in-action/java/src/main/java/ingestion/database/dialect/InformixToDatasetApp.java
package ingestion.database.dialect;

import org.apache.spark.sql.Dataset;
import org.apache.spark.sql.Row;
import org.apache.spark.sql.SparkSession;
import org.apache.spark.sql.jdbc.JdbcDialect;
import org.apache.spark.sql.jdbc.JdbcDialects;

public class InformixToDatasetApp {

    public static void main(String[] args) {
        InformixToDatasetApp app = new InformixToDatasetApp();
        app.start();
    }

    private void start() {
        SparkSession spark = SparkSession.builder()
                .appName("Informix to Dataframe using a JDBC Connection")
                .master("local")
                .getOrCreate();

        spark.sparkContext().setLogLevel("ERROR");

        // Specific informix dialect
        JdbcDialect dialect = new InformixJdbcDialect();
        JdbcDialects.registerDialect(dialect);

        Dataset<Row> df = spark.read()
                .format("jdbc")
                .option(
                        "url",
                        "jdbc:informix-sqli://[::1]:33378/stores_demo:IFXHOST=lo_informix1210;DELIMIDENT=Y")
                .option("dbtable", "customer")
                .option("user", "informix")
                .option("password", "informix")
                .load();

        df.show(5);
        df.printSchema();
        System.out.println("the dataframe contains " + df.count() + " records");

        spark.stop();
    }
}

File: spark-in-action/scala/src/main/scala/ingestion/database/dialect/InformixToDatasetApp.scala
package ingestion.database.dialect

import org.apache.spark.sql.SparkSession
import org.apache.spark.sql.jdbc.JdbcDialects

object InformixToDatasetApp {

  def main(args: Array[String]): Unit = {
    val spark: SparkSession = SparkSession.builder
      .appName("Informix to Dataframe using a JDBC Connection")
      .master("local[*]")
      .getOrCreate

    val dialect = new InformixJdbcDialect
    JdbcDialects.registerDialect(dialect)

    val informixURL = "jdbc:informix-sqli://[::1]:33378/stores_demo:IFXHOST=lo_informix1210;DELIMIDENT=Y"

    // Using properties
    val df = spark.read
      .format("jdbc")
      .option("url", informixURL)
      .option("dbtable", "customer")
      .option("user", "informix")
      .option("password", "informix")
      .load

    df.show(5)
    df.printSchema
    
    spark.stop
  }

}
